packages/nvidia_nat_nemo_customizer/src/nat/plugins/customizer/dpo/trainer_adapter.py

```
@@ -400,13 +400,59 @@ async def wait_until_complete(self, ref: TrainingJobRef, poll_interval: float |
+                # Clean up active job tracking first
                 self._active_jobs.pop(ref.run_id, None)
 
+                # Handle non-successful completions with clear error messages
+                if status.status == TrainingStatusEnum.FAILED:
+                    error_msg = status.message or "Training job failed"
+                    logger.error(f"Training job {ref.run_id} FAILED: {error_msg}")
+                    raise RuntimeError(
+                        f"Training job {ref.run_id} failed: {error_msg}. "
+                        f"Model was not trained and cannot be deployed. "
+                        f"Please check the job logs for details and retry the training."
+                    )
```

Ruff (0.14.10)

Avoid specifying long messages outside the exception class

(TRY003)